### PR TITLE
39: Add internal name for field + use this informations in fields selector.

### PR DESCRIPTION
**Pull request by arimet**
## Screenshot

image
image
image

### CODE REVIEW

File: src/app/js/fields/FieldInternalState.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+const useStyles = makeStyles({
+    container: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+        marginRight: '1rem',
+    },
+    state: {
+        padding: '0.5rem',
+        border: '1px solid black',
```
Comment (guilbill) [inline src/app/js/fields/FieldInternalState.js]: I've noticed that the black is not the right color, I think the color should be theme.black.secondary

**Review by guilbill — CHANGES_REQUESTED** [review]
Well done for the "toggle" button !

Although I wonder why not having "Toggle" in the component name. One more thing to note is that I think it's a bad idea to have "state" in the component name also (state already have a meaning in React, etc.), maybe... Scope? (as it is how this info is named in the code as far as I understand...)

File: src/app/js/fields/FieldToggleInternalScope.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+export const FieldToggleInternalScopeComponent = ({ input }) => {
+    const classes = useStyles();
+
+    const handleStateSelected = (event, state) => {
+        input.onChange(state === input.value ? '' : state);
+    };
+
+    return (
+        <ToggleButtonGroup
```
Comment (guilbill) [inline src/app/js/fields/FieldToggleInternalScope.js:25]: Nice

File: src/app/js/fields/FieldInternalIcon.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+INTERNAL_STATE_ICON['document'] = <DescriptionIcon />;
+INTERNAL_STATE_ICON['chart'] = <BarChartIcon />;
+
+const getIconInternalScope = state => {
+    if (!!state) {
+        return INTERNAL_STATE_ICON[state];
+    }
+};
+
+const FieldInternalIcon = ({ state }) => (getIconInternalScope(state));
```
Comment (guilbill) [inline src/app/js/fields/FieldInternalIcon.js]: Is it still in use?

**Review by guilbill — CHANGES_REQUESTED** [review]

File: src/app/js/fields/FieldInternalIcon.js (hunk 3 above)
Comment (arimet) [reply src/app/js/fields/FieldInternalIcon.js]: It is still in use,  it is the component which returns the good icon in the selection of the fields and in the datagrid. but the term state doesn't make sense anymore, I just changed it to scope

**Review by guilbill — APPROVED** [review]

**Pull request merged**